doc/login.md

```diff
@@ -0,0 +1,40 @@
+
+Cockpit Login
+================================
+
+The cockpit login page signs users in by sending requests to /login the a Authorization header.
+Generally this will be with a user provided username and password. In that case the authorization
+header is built like this.
+
+```
+Basic base64(user:password)
+```
+
+
+A successful response is a 200 http code with a json body that contains a ```user``` field with the user
+name of the user that was just logged in. Additional fields may be present
+
+Other http codes are considered errors. Generally these are 401 or 403 http codes
+and should contain a json body with the following fields:
+
+ * error: Values of ```authentication-failed```, ```authentication-unavailable``` or ```permission-denied``` are translated to the appropriate cockpit error codes. Any other values are treated as generic errors
+ * message: Optional text with more details about the error.
+
+In some authentication setups addtional steps are required. When this happens
+the json object will have a prompt and a id field.
```

Comment:
Have you considered sending back the challenge in headers? Symmetric to the reply? Willing to discuss this. But I'd like to make sure we're getting this right.